### PR TITLE
test: add tags with git branch/commit information to sample apps

### PR DESCRIPTION
**Pull request by armcknight**
For https://github.com/getsentry/sentry-cocoa/issues/4321

Add commit and branch info. Will attach `-dirtied` to commit hash if there are changes to the working index. If working from a detached head state, the branch is reported as `HEAD`.

Dirty index on a branch: https://sentry-sdks.sentry.io/issues/5485500008/events/latest/?project=5428557&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=latest-event&statsPeriod=1h&stream_index=0

image

Detached HEAD with clean index: https://sentry-sdks.sentry.io/issues/5485405120/events/latest/?project=5428557&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=latest-event&statsPeriod=1m&stream_index=0

image

#skip-changelog

### CODE REVIEW

**Review by brustolin — COMMENTED** [review]
Thanks for this.

I’m just missing automated integration with the CI, so that when the UI tests run, we have this information included in the events.

**Comment by armcknight** [comment]
What makes you think this wouldn't work in CI? It's part of the app build, so it's going to happen every time the app is built, which happens in CI before running the tests. I just reran the iOS-Swift UI tests, here is an example.
image
https://sentry-sdks.sentry.io/issues/5485405130/events/latest/?project=5428557&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=latest-event&statsPeriod=1h&stream_index=3

**Comment by armcknight** [comment]
This does still need to be added for everything other than iOS-Swift and iOS13-Swift

File: Samples/iOS-Swift/iOS13-Swift/Info.plist [hunk 1]
```diff
@@ -2,6 +2,12 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>GIT_BRANCH</key>
+	<string>armcknight/ci/inject-git-branch-commit-hash</string>
```
Comment (bruno-garcia) [inline Samples/iOS-Swift/iOS13-Swift/Info.plist:6]: This was a test that needs to be deleted?

**Review by bruno-garcia — APPROVED** [review]
Awesome stuff!!!

**Comment by brustolin** [comment]
> What makes you think this wouldn't work in CI?

Sorry, I skipped `project.pbxproj` you're using the script during build phase.
All good.

**Review by brustolin — APPROVED** [review]

**Comment by armcknight** [comment]
I added it to the other test projects i could, with some exceptions, and tested them all to make sure they work. see the samples readme for more details

**Pull request merged**